tck/src/main/java/ee/jakarta/tck/data/framework/read/only/ReadOnlyRepository.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package ee.jakarta.tck.data.framework.read.only;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import jakarta.data.repository.DataRepository;
+
+//FIXME - Are user defined repository interfaces like this allowed via the Specification? 
```

Comment:
Good question.  I'm guessing that you are asking about repository interfaces with type parameters for the entity and id, but which don't inherit from a built-in interface like DataRepository or CrudRepository. The spec does let you use your own interface, but doesn't clearly define for that case how the entity type and id type are determined, although it makes sense to infer them from the type parameters in the same way as built-in interfaces.  Probably we just need to add the extra clarity for this case, but until then, it should be avoided in the TCK and the well-defined approach used instead.